cpp/daal/src/algorithms/kmeans/oneapi/cl_kernels/kmeans_cl_kernels.cl

```diff
@@ -147,7 +147,8 @@ DECLARE_SOURCE(
         int minIdx             = -1;
```

Comment:
line 174: you are declaring variable `HUGE` in two functions of this file. But its name is written like a global constant variable. I suggest giving lowercase and more meaningful names for these variables to avoid confusion.
Moreover, on line 231 you are assigning to it a value of `-1.0e-15`, which does not seem huge at all for me